docs/javascript-plugins.md

```diff
@@ -0,0 +1,122 @@
+---
+title: JavaScript Plugins
+description: Plugins implemented in JavaScript are easily used across different projects, as they do not require any additional programs to be installed.
+---
+
+JavaScript plugins are native MyST plugins, which are loaded as modules into the MyST engine. Transforms defined in these modules have access to helpful AST manipulation routines made available by MyST. Edits to JavaScript plugins have no effect during execution of a MyST build, instead the build must be restarted.
+
+## Defining a new directive
+
+To create a plugin, you will need a single Javascript file[^esm] that exports one or more of the objects above. For example, a simple directive that pulls a random image from [Unsplash](https://unsplash.com/) can be created with a single file that exports an `unsplash` directive.
+
+[^esm]: The format of the Javascript should be an ECMAScript modules, not CommonJS. This means it uses `import` statements rather than `require()` and is the most modern style of Javascript.
+
+:::{literalinclude} unsplash.mjs
+:name: unsplash-js-source
+:caption: A plugin to add an `unsplash` directive that includes a beautiful, random picture based on a query string.
+:::
+
+This code should be referenced from your `myst.yml` under the `projects.plugins`:
+
+```{code} yaml
+:filename: myst.yml
+project:
+  plugins:
+    - type: javascript
+      path: unsplash.mjs
```

Comment:
Is it possible to leave things as:

```suggestion
    - unsplash.mjs
```

And it gets transformed?

Reply:
Perhaps having a small comment about that? Maybe in tabs?

Reply:
It is. I suppose we can keep JS plugins as the "preferred route" and error if the ext isn't mjs.